RHEL6 failure - e2e OCI test needs overlay

### Version of Singularity:

What version of Singularity are you using? Run:

```
3.5.2
```

### Actual behavior

```
--- FAIL: TestE2E/PAR/OCI/basic (0.25s)
                singularitycmd.go:653: Running command "/usr/local/bin/singularity --debug oci mount /tmp/stest.490211207/test.sif /tmp/stest.490211207/bundle-728097247"
                singularitycmd.go:653: 
                    "/usr/local/bin/singularity --debug oci mount /tmp/stest.490211207/test.sif /tmp/stest.490211207/bundle-728097247" output:
                    ESC[0mDEBUG   ESC[0m[U=0,P=16049]      persistentPreRunE()           Singularity version: 3.5.2+141-g7742e82
                    ESC[0mDEBUG   ESC[0m[U=0,P=16049]      handleConfDir()               /root/.singularity already exists. Not creating.
                    ESC[0mDEBUG   ESC[0m[U=0,P=16049]      Init()                        Image format detection
                    ESC[0mDEBUG   ESC[0m[U=0,P=16049]      Init()                        Check for sandbox image format
                    ESC[0mDEBUG   ESC[0m[U=0,P=16049]      Init()                        sandbox format initializer returned: not a directory image
                    ESC[0mDEBUG   ESC[0m[U=0,P=16049]      Init()                        Check for sif image format
                    ESC[0mDEBUG   ESC[0m[U=0,P=16049]      Init()                        sif image format detected
                    ESC[31mFATAL   ESC[0m[U=0,P=16049]      func39()                      failed to create overlay: failed to mount /tmp/stest.490211207/bundle-728097247/overlay: no such device
                    
                singularitycmd.go:653: got 255 as exit code and was expecting 0: exit status 255
                    waiting for command "/usr/local/bin/singularity --debug oci mount /tmp/stest.490211207/test.sif /tmp/stest.490211207/bundle-728097247"
                    github.com/sylabs/singularity/e2e/internal/e2e.TestEnv.RunSingularity.func1
                        github.com/sylabs/singularity@/e2e/internal/e2e/singularitycmd.go:628
                    github.com/sylabs/singularity/e2e/internal/e2e.Privileged.func1
                        github.com/sylabs/singularity@/e2e/internal/e2e/priv_linux.go:53
                    github.com/sylabs/singularity/e2e/internal/e2e.TestEnv.RunSingularity.func2
                        github.com/sylabs/singularity@/e2e/internal/e2e/singularitycmd.go:653
                    runtime.goexit
                        runtime/asm_amd64.s:1357
                oci.go:65: failed to mount OCI bundle image
                singularitycmd.go:653: Running command "/usr/local/bin/singularity --debug oci create -b /tmp/stest.490211207/bundle-728097247 d20620e0-55fd-4557-a03e-af90bec89432"
                singularitycmd.go:653: Running command "/usr/local/bin/singularity --debug oci start d20620e0-55fd-4557-a03e-af90bec89432"
                require.go:46: cgroups disabled
                require.go:46: cgroups disabled
                singularitycmd.go:653: Running command "/usr/local/bin/singularity --debug oci start d20620e0-55fd-4557-a03e-af90bec89432"
                singularitycmd.go:653: Running command "/usr/local/bin/singularity --debug oci exec d20620e0-55fd-4557-a03e-af90bec89432 id"
                singularitycmd.go:653: Running command "/usr/local/bin/singularity --debug oci kill -t 2 d20620e0-55fd-4557-a03e-af90bec89432 KILL"
                singularitycmd.go:653: Running command "/usr/local/bin/singularity --debug oci delete d20620e0-55fd-4557-a03e-af90bec89432"
                singularitycmd.go:653: Running command "/usr/local/bin/singularity --debug oci state d20620e0-55fd-4557-a03e-af90bec89432"
                singularitycmd.go:653: Running command "/usr/local/bin/singularity --debug oci kill d20620e0-55fd-4557-a03e-af90bec89432"
                singularitycmd.go:653: Running command "/usr/local/bin/singularity --debug oci start d20620e0-55fd-4557-a03e-af90bec89432"
                singularitycmd.go:653: Running command "/usr/local/bin/singularity --debug oci umount /tmp/stest.490211207/bundle-728097247"
```

### What OS/distro are you running

```
RHEL6
```

--- a/e2e/actions/actions.go
+++ b/e2e/actions/actions.go
@@ -486,6 +486,7 @@ func (c actionTests) RunFromURI(t *testing.T) {
 		command string
 		argv    []string
 		exit    int
+		profile e2e.Profile
 	}{
 		// Run from supported URI's and check the runscript call works
 		{
@@ -493,12 +494,14 @@ func (c actionTests) RunFromURI(t *testing.T) {
 			command: "run",
 			argv:    []string{"--bind", bind, "docker://busybox:latest", size},
 			exit:    0,
+			profile: e2e.UserProfile,
 		},
 		{
 			name:    "RunFromLibraryOK",
 			command: "run",
 			argv:    []string{"--bind", bind, "library://busybox:latest", size},
 			exit:    0,
+			profile: e2e.UserProfile,
 		},
 		// TODO(mem): reenable this; disabled while shub is down
 		// {
@@ -506,24 +509,28 @@ func (c actionTests) RunFromURI(t *testing.T) {
 		// 	command: "run",
 		// 	argv:    []string{"--bind", bind, "shub://singularityhub/busybox", size},
 		// 	exit:    0,
+		// 	profile: e2e.UserProfile,
 		// },
 		{
 			name:    "RunFromOrasOK",
 			command: "run",
 			argv:    []string{"--bind", bind, c.env.OrasTestImage, size},
 			exit:    0,
+			profile: e2e.UserProfile,
 		},
 		{
 			name:    "RunFromDockerKO",
 			command: "run",
 			argv:    []string{"--bind", bind, "docker://busybox:latest", "0"},
 			exit:    1,
+			profile: e2e.UserProfile,
 		},
 		{
 			name:    "RunFromLibraryKO",
 			command: "run",
 			argv:    []string{"--bind", bind, "library://busybox:latest", "0"},
 			exit:    1,
+			profile: e2e.UserProfile,
 		},
 		// TODO(mem): reenable this; disabled while shub is down
 		// {
@@ -531,12 +538,14 @@ func (c actionTests) RunFromURI(t *testing.T) {
 		// 	command: "run",
 		// 	argv:    []string{"--bind", bind, "shub://singularityhub/busybox", "0"},
 		// 	exit:    1,
+		// 	profile: e2e.UserProfile,
 		// },
 		{
 			name:    "RunFromOrasKO",
 			command: "run",
 			argv:    []string{"--bind", bind, c.env.OrasTestImage, "0"},
 			exit:    1,
+			profile: e2e.UserProfile,
 		},
 
 		// exec from a supported URI's and check the exit code
@@ -545,12 +554,14 @@ func (c actionTests) RunFromURI(t *testing.T) {
 			command: "exec",
 			argv:    []string{"docker://busybox:latest", "true"},
 			exit:    0,
+			profile: e2e.UserProfile,
 		},
 		{
 			name:    "ExecTrueLibrary",
 			command: "exec",
 			argv:    []string{"library://busybox:latest", "true"},
 			exit:    0,
+			profile: e2e.UserProfile,
 		},
 		// TODO(mem): reenable this; disabled while shub is down
 		// {
@@ -558,24 +569,28 @@ func (c actionTests) RunFromURI(t *testing.T) {
 		// 	command: "exec",
 		// 	argv:    []string{"shub://singularityhub/busybox", "true"},
 		// 	exit:    0,
+		// 	profile: e2e.UserProfile,
 		// },
 		{
 			name:    "ExecTrueOras",
 			command: "exec",
 			argv:    []string{c.env.OrasTestImage, "true"},
 			exit:    0,
+			profile: e2e.UserProfile,
 		},
 		{
 			name:    "ExecFalseDocker",
 			command: "exec",
 			argv:    []string{"docker://busybox:latest", "false"},
 			exit:    1,
+			profile: e2e.UserProfile,
 		},
 		{
 			name:    "ExecFalseLibrary",
 			command: "exec",
 			argv:    []string{"library://busybox:latest", "false"},
 			exit:    1,
+			profile: e2e.UserProfile,
 		},
 		// TODO(mem): reenable this; disabled while shub is down
 		// {
@@ -583,64 +598,74 @@ func (c actionTests) RunFromURI(t *testing.T) {
 		// 	command: "exec",
 		// 	argv:    []string{"shub://singularityhub/busybox", "false"},
 		// 	exit:    1,
+		// 	profile: e2e.UserProfile,
 		// },
 		{
 			name:    "ExecFalseOras",
 			command: "exec",
 			argv:    []string{c.env.OrasTestImage, "false"},
 			exit:    1,
+			profile: e2e.UserProfile,
 		},
 
 		// exec from URI with user namespace enabled
 		{
 			name:    "ExecTrueDockerUserns",
 			command: "exec",
-			argv:    []string{"--userns", "docker://busybox:latest", "true"},
+			argv:    []string{"docker://busybox:latest", "true"},
 			exit:    0,
+			profile: e2e.UserNamespaceProfile,
 		},
 		{
 			name:    "ExecTrueLibraryUserns",
 			command: "exec",
-			argv:    []string{"--userns", "library://busybox:latest", "true"},
+			argv:    []string{"library://busybox:latest", "true"},
 			exit:    0,
+			profile: e2e.UserNamespaceProfile,
 		},
 		// TODO(mem): reenable this; disabled while shub is down
 		// {
 		// 	name:    "ExecTrueShubUserns",
 		// 	command: "exec",
-		// 	argv:    []string{"--userns", "shub://singularityhub/busybox", "true"},
+		// 	argv:    []string{"shub://singularityhub/busybox", "true"},
 		// 	exit:    0,
+		// 	profile: e2e.UserNamespaceProfile,
 		// },
 		{
 			name:    "ExecTrueOrasUserns",
 			command: "exec",
-			argv:    []string{"--userns", c.env.OrasTestImage, "true"},
+			argv:    []string{c.env.OrasTestImage, "true"},
 			exit:    0,
+			profile: e2e.UserNamespaceProfile,
 		},
 		{
 			name:    "ExecFalseDockerUserns",
 			command: "exec",
-			argv:    []string{"--userns", "docker://busybox:latest", "false"},
+			argv:    []string{"docker://busybox:latest", "false"},
 			exit:    1,
+			profile: e2e.UserNamespaceProfile,
 		},
 		{
 			name:    "ExecFalseLibraryUserns",
 			command: "exec",
-			argv:    []string{"--userns", "library://busybox:latest", "false"},
+			argv:    []string{"library://busybox:latest", "false"},
 			exit:    1,
+			profile: e2e.UserNamespaceProfile,
 		},
 		// TODO(mem): reenable this; disabled while shub is down
 		// {
 		// 	name:    "ExecFalseShubUserns",
 		// 	command: "exec",
-		// 	argv:    []string{"--userns", "shub://singularityhub/busybox", "false"},
+		// 	argv:    []string{"shub://singularityhub/busybox", "false"},
 		// 	exit:    1,
+		// 	profile: e2e.UserNamespaceProfile,
 		// },
 		{
 			name:    "ExecFalseOrasUserns",
 			command: "exec",
-			argv:    []string{"--userns", c.env.OrasTestImage, "false"},
+			argv:    []string{c.env.OrasTestImage, "false"},
 			exit:    1,
+			profile: e2e.UserNamespaceProfile,
 		},
 	}
 
@@ -648,7 +673,7 @@ func (c actionTests) RunFromURI(t *testing.T) {
 		c.env.RunSingularity(
 			t,
 			e2e.AsSubtest(tt.name),
-			e2e.WithProfile(e2e.UserProfile),
+			e2e.WithProfile(tt.profile),
 			e2e.WithCommand(tt.command),
 			e2e.WithArgs(tt.argv...),
 			e2e.ExpectExit(tt.exit),

--- a/e2e/oci/oci.go
+++ b/e2e/oci/oci.go
@@ -48,6 +48,7 @@ func (c *ctx) checkOciState(t *testing.T, containerID, state string) {
 }
 
 func genericOciMount(t *testing.T, c *ctx) (string, func()) {
+	require.Filesystem(t, "overlay")
 	bundleDir, err := ioutil.TempDir(c.env.TestDir, "bundle-")
 	if err != nil {
 		err = errors.Wrapf(err, "creating temporary bundle directory at %q", c.env.TestDir)

